server/server.cpp

```diff
@@ -1,174 +1,166 @@
-#include <iostream>
 #include <fstream>
+#include <iostream>
 #include <thread>
 
 #include <components.pb.h>
 #include <google/protobuf/util/json_util.h>
 
-#include "server.hpp"
 #include "client-connection.hpp"
 #include "filesystem.hpp"
+#include "server.hpp"
 
 using asio::ip::tcp;
 
 namespace tec {
-	namespace networking {
-		unsigned short PORT = 0xa10c;
-		std::mutex Server::recent_msgs_mutex;
-
-		Server::Server(tcp::endpoint& endpoint) : acceptor(io_service, endpoint), socket(io_service) {
-
-			// Create a simple greeting chat message that all clients get.
-			std::string message("Hello from server\n");
-			this->greeting_msg.SetBodyLength(message.size());
-			memcpy(this->greeting_msg.GetBodyPTR(), message.c_str(), this->greeting_msg.GetBodyLength());
-			this->greeting_msg.encode_header();
+namespace networking {
+unsigned short PORT = 0xa10c;
+std::mutex Server::recent_msgs_mutex;
+
+Server::Server(tcp::endpoint& endpoint) : acceptor(io_service, endpoint), socket(io_service) {
+	// Create a simple greeting chat message that all clients get.
+	std::string message("Hello from server\n");
+	this->greeting_msg.SetBodyLength(message.size());
+	memcpy(this->greeting_msg.GetBodyPTR(), message.c_str(), this->greeting_msg.GetBodyLength());
+	this->greeting_msg.encode_header();
+
+	AcceptHandler();
+}
 
-			AcceptHandler();
-		}
-
-		// TODO: Implement a method to deliver a message to all clients except the source.
-		void Server::Deliver(const ServerMessage& msg, bool save_to_recent) {
-			if (save_to_recent) {
-				std::lock_guard<std::mutex> lg(recent_msgs_mutex);
-				this->recent_msgs.push_back(msg);
-				while (this->recent_msgs.size() > max_recent_msgs) {
-					this->recent_msgs.pop_front();
-				}
-			}
-
-			{
-				std::lock_guard<std::mutex> lg(client_list_mutex);
-				for (auto client : this->clients) {
-					client->QueueWrite(msg);
-				}
-			}
+// TODO: Implement a method to deliver a message to all clients except the source.
+void Server::Deliver(const ServerMessage& msg, bool save_to_recent) {
+	if (save_to_recent) {
+		std::lock_guard<std::mutex> lg(recent_msgs_mutex);
+		this->recent_msgs.push_back(msg);
+		while (this->recent_msgs.size() > max_recent_msgs) {
+			this->recent_msgs.pop_front();
 		}
+	}
 
-		void Server::Deliver(std::shared_ptr<ClientConnection> client, const ServerMessage& msg) {
+	{
```

Comment:
Inner scope is redundant as the lock_guard goes out of scope immediately anyways.

Reply:
I kind of want to directly work on the networking layer next, so you can probably leave it if you haven't changed it already.

Reply:
I did remove the inner scope, but that is it.